### PR TITLE
Task 7

### PR DESCRIPTION
**Pull request by unfalse**
**Main:**
1 - authorization-service is added to the repo, has correct basicAuthorizer lambda and correct serverless.yaml file
3 - import-service serverless.yaml file has authorizer configuration for the importProductsFile lambda. Request to the importProductsFile lambda should work only with correct authorization_token being decoded and checked by basicAuthorizer lambda. Response should be in 403 HTTP status if access is denied for this user (invalid authorization_token) and in 401 HTTP status if Authorization header is not provided.
5 - update client application to send Authorization: Basic authorization_token header on import. Client should get authorization_token value from browser localStorage (authorization_token = localStorage.getItem('authorization_token'))

**Additional:**
+1 - Client application should display alerts for the responses in 401 and 403 HTTP statuses. This behavior should be added to the nodejs-aws-fe-main/src/index.tsx file

Add Login page and protect getProducts lambda by the Cognito Authorizer (no evaluation):
Login page: https://unfalse-task7-pool.auth.eu-west-1.amazoncognito.com/login?client_id=5tm6novklipcbis16pqabmh54e&response_type=token&scope=aws.cognito.signin.user.admin+email+openid+phone+profile&redirect_uri=https://d2p1ss31jpj477.cloudfront.net/

- Cloudfront: https://d2p1ss31jpj477.cloudfront.net/admin/products
- FE PR: https://github.com/unfalse/node-js-aws-fe-shop/pull/6

### CODE REVIEW

File: authorization-service/src/utils/policy.ts [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+          Resource: resource
+        }
+      ]
+    }
+  });
+
+export const getCredentialsFromToken = (authorizationToken: string) => {
+  const [, encodedCreds = ''] = authorizationToken.split(' ');
+  const buff = Buffer.from(encodedCreds, 'base64');
+  const [username = '', password = ''] = buff.toString('utf-8').split(':');
```
Comment (roman-shuvalov) [inline authorization-service/src/utils/policy.ts:18]: Wrong token format!
Expected:
unfalse=TEST_PASSWORD
Actual:
unfalse:TEST_PASSWORD

Comment (unfalse) [reply authorization-service/src/utils/policy.ts:18]: I disagree. Please read carefully the task:
_This lambda should have at least one environment variable with credentials: {yours_github_account_login}=TEST_PASSWORD_
and
_Basic Authorization header: Authorization: Basic authorization_token where authorization_token is equal base64-encoded({yours_github_account_login}:TEST_PASSWORD)_

See the difference?

File: import-service/src/functions/importProductsFile/index.ts [hunk 2]
```diff
@@ -14,6 +14,13 @@ export const importProductsFile = {
               name: true
             }
           }
+        },
+        authorizer: {
+          name: 'basicAuthorizer',
+          arn: 'arn:aws:lambda:eu-west-1:030019190562:function:authorization-service-dev-basicAuthorizer',
```
Comment (serguntchik) [inline import-service/src/functions/importProductsFile/index.ts]: Небольшое замечание, не влияющее на итоговые баллы :) Хардкодить арны не очень правильно, гораздо лучше сделать экспорт этого арна из сервиса авторизации и заимпортить в импорт-сервисе по имени

Comment (unfalse) [reply import-service/src/functions/importProductsFile/index.ts]: знаю, торопился. Попробую сделать.

Comment (unfalse) [reply import-service/src/functions/importProductsFile/index.ts]: добавил экспорт

**Pull request merged**